crates/common/download/src/download.rs

```diff
@@ -559,7 +559,7 @@ mod tests {
     async fn downloader_download_with_content_length_larger_than_usable_disk_space() {
         use nix::sys::statvfs;
         let tmpstats = statvfs::statvfs("/tmp").unwrap();
-        let usable_disk_space = tmpstats.blocks_free() * tmpstats.block_size();
+        let usable_disk_space = (tmpstats.blocks_free() as u64) * (tmpstats.block_size() as u64);
```

Comment:
This test was failing to compile for some 32 bit targets as it seems that the two types returned from the function calls would depend on the target.

Reply:
As the type of the value returned by `tmpstats.blocks_free()` is target dependent it's better to use `u64::from()` instead of `as 64`, even if this is more than unlikely that a target would use a type larger than `u64`.

```suggestion
        let usable_disk_space = u64::from(tmpstats.blocks_free()) * u64::from(tmpstats.block_size());
```

Reply:
Though the suggested change invoke the following clippy warning:
<img width="1145" alt="image" src="https://github.com/thin-edge/thin-edge.io/assets/3029781/c25d6316-c471-48ae-9f62-6d3f89f6715c">

Reply:
So let's go using `as u64`.